src/components/MultiHopTrade/components/RateGasRow.tsx

```diff
@@ -1,41 +1,125 @@
+import { ArrowUpDownIcon, ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons'
 import type { FlexProps } from '@chakra-ui/react'
-import { Stack } from '@chakra-ui/react'
-import { type FC, memo } from 'react'
+import {
+  Box,
+  Center,
+  Collapse,
+  Flex,
+  Skeleton,
+  Stack,
+  Tooltip,
+  useDisclosure,
+} from '@chakra-ui/react'
+import type { SwapperName, SwapSource } from '@shapeshiftoss/swapper'
+import { AnimatePresence } from 'framer-motion'
+import type { PropsWithChildren } from 'react'
+import { type FC, memo, useMemo } from 'react'
 import { FaGasPump } from 'react-icons/fa'
 import { useTranslate } from 'react-polyglot'
 import { Amount } from 'components/Amount/Amount'
-import { CircularProgress } from 'components/CircularProgress/CircularProgress'
 import { HelperTooltip } from 'components/HelperTooltip/HelperTooltip'
+import { StreamIcon } from 'components/Icons/Stream'
 import { Row } from 'components/Row/Row'
+import { SlideTransitionX } from 'components/SlideTransitionX'
 import { Text } from 'components/Text'
 import { bnOrZero } from 'lib/bignumber/bignumber'
 import { firstNonZeroDecimal } from 'lib/math'
+import {
+  THORCHAIN_LONGTAIL_STREAMING_SWAP_SOURCE,
+  THORCHAIN_STREAM_SWAP_SOURCE,
+} from 'lib/swapper/swappers/ThorchainSwapper/constants'
+
+import { SwapperIcon } from './TradeInput/components/SwapperIcon/SwapperIcon'
 
 type RateGasRowProps = {
   sellSymbol?: string
   buySymbol?: string
   rate?: string
   gasFee: string
   isLoading?: boolean
-  isError?: boolean
-}
+  allowSelectQuote: boolean
+  swapperName?: SwapperName
+  swapSource?: SwapSource
+  onRateClick?: () => void
+} & PropsWithChildren
 
 const helpersTooltipFlexProps: FlexProps = { flexDirection: 'row-reverse' }
+const rowHover = { bg: 'background.surface.raised.base' }
+const rateHover = {
+  cursor: 'pointer',
+  '.rate': { borderColor: 'text.link' },
+}
 
 export const RateGasRow: FC<RateGasRowProps> = memo(
-  ({ sellSymbol, buySymbol, rate, gasFee, isLoading, isError }) => {
+  ({
+    sellSymbol,
+    buySymbol,
+    rate,
+    gasFee,
+    isLoading,
+    allowSelectQuote,
+    swapperName,
+    swapSource,
+    onRateClick,
+    children,
+  }) => {
     const translate = useTranslate()
+    const { isOpen, onToggle } = useDisclosure()
+
+    const swapperIcons = useMemo(() => {
+      const isStreaming =
+        swapSource === THORCHAIN_STREAM_SWAP_SOURCE ||
+        swapSource === THORCHAIN_LONGTAIL_STREAMING_SWAP_SOURCE
```

Comment:
Since we often seem to forget the streaming swap source, do we want to abstract this into a common guard/util?

Reply:
can do in follow-up 👍